### PR TITLE
chore(headless): initial work to cleanup headless bundles

### PR DESCRIPTION
**Pull request by olamothe**
So I've done quite a bit of testing around the work that will be necessary to make our bundles leaner.

One thing I've realized is everything around three shaking will require quite a lot of manual work to help bundle tools (esbuild, webpack, rollup)  identify functions/modules/classes that do not have side effects ( https://esbuild.github.io/api/#tree-shaking // https://esbuild.github.io/api/#pure ). Right now, I'd say only about 5 to 10% of Headless package is tree shakable, which is far lower that what is should be.

In the first phase, what I'd like to achieve is to simply changes the way we do imports so that we can shake off some weights from each produced bundles, notwithstanding any tree shaking capabilities.

This in itself will be a lot of work, as we need to juggle some imports that are used in hundreds of files.

This PR contains: 
* Adding "metafile" when we build packages. These files contains information about how esbuild built the project. This is what I've been using to manually identify the import chain so I can start refactoring how we do imports internally. I've put them in the `dist` bundle, which means we would now ship those. This can be helpful I think, to add tooling down the line around all of this, when we do a release. Downside is that we add weights to npm install. An alternative would be to put them in a different folder, which we could git ignore. Open to any idea you guys would have.

* Starting to do some minor import changes, in order to remove `productRecommendation` use case from `searchPage` use case.

https://coveord.atlassian.net/browse/KIT-2452

### CODE REVIEW

File: packages/headless/esbuild.mjs [hunk 1]
```diff
@@ -131,39 +141,49 @@ function buildBrowserConfig(options) {
     ],
     ...options,
   });
+  outputMetafile(`browser.${options.format}`, outDir, out.metafile);
```
Comment (btaillon) [inline packages/headless/esbuild.mjs:144]: Similarly to the Stencil definitions file, I think it may be useful to include the metafile in Git/PRs. It looks like it would make it easy to identify changes in the bundle at a pretty low effort cost.

File: packages/headless/src/features/configuration/search-configuration-actions-loader.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import {PayloadAction} from '@reduxjs/toolkit';
 import {RecommendationEngine} from '../../app/recommendation-engine/recommendation-engine';
-import {configuration, pipeline, searchHub} from '../../app/reducers';
+import {configuration} from '../../app/reducers';
+import {pipeline, searchHub} from '../../app/reducers';
```
Comment (btaillon) [inline packages/headless/src/features/configuration/search-configuration-actions-loader.ts:4]: Did you mean to split those?

Suggestion to revert the change:
```suggestion
import {configuration, pipeline, searchHub} from '../../app/reducers';
```

**Review by btaillon — APPROVED** [review]

File: packages/headless/esbuild.mjs (hunk 1 above)
Comment (olamothe) [reply packages/headless/esbuild.mjs:144]: I want to work on this at a later stage: But yes, eventually I want to add reports directly in PR to see changes. 

I've explored some tools that will allow us to do this (eg: https://esbuild.github.io/analyze/)
